### PR TITLE
refactor: split bank structure to support multiple variants

### PR DESCRIPTION
**Pull request by kashif-m**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Bank structure for payouts is converted to an enum to support multiple variants - ACH / Bacs / Sepa
- Vault will still use a single implementation for BankPayout where all the fields are optional
- Changes for the same are accommodated in Adyen and Wise integrations

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Tested this locally via Postman on both Adyen and Wise.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/api_models/src/payouts.rs [hunk 1]
```diff
@@ -163,37 +163,66 @@ pub struct Card {
...
+    /// Bank city
+    #[schema(value_type = String, example = "California")]
+    pub bank_city: String,
 
-    /// Bankleitzahl [blz] - used in Germany and Austria for identifying a bank and it's branches.
-    #[schema(value_type = String, example = "10070024")]
-    pub blz: Option<String>,
+    /// Bank account number is an unique identifier assigned by a bank to a customer.
+    #[schema(value_type = String, example = "000123456")]
+    pub bank_account_number: String,
```
Comment (jagan-jaya) [inline crates/api_models/src/payouts.rs:215]: mask it with secret

File: crates/api_models/src/payouts.rs [hunk 2]
```diff
@@ -163,37 +163,66 @@ pub struct Card {
...
 
-    /// International Bank Account Number (iban) - used in many countries for identifying a bank along with it's customer.
-    #[schema(value_type = String, example = "DE89370400440532013000")]
-    pub iban: Option<String>,
+#[cfg(feature = "payouts")]
+#[derive(Default, Eq, PartialEq, Clone, Debug, Deserialize, Serialize, ToSchema)]
+pub struct BacsBankTransfer {
+    /// Bank name
+    #[schema(value_type = String, example = "Deutsche Bank")]
+    pub bank_name: String,
```
Comment (jagan-jaya) [inline crates/api_models/src/payouts.rs:203]: mask it with secret

File: crates/api_models/src/payouts.rs [hunk 3]
```diff
@@ -205,110 +234,14 @@ pub struct Bank {
...
-            bank_transit_number: None,
-            bank_name: "Deutsche Bank".to_string(),
-            bank_country_code: api_enums::CountryAlpha2::US,
-            bank_city: "California".to_string(),
-        }
-    }
-}
+    /// International Bank Account Number (iban) - used in many countries for identifying a bank along with it's customer.
+    #[schema(value_type = String, example = "DE89370400440532013000")]
+    pub iban: String,
```
Comment (jagan-jaya) [inline crates/api_models/src/payouts.rs:240]: mask it with secret

File: crates/api_models/src/payment_methods.rs [hunk 4]
```diff
@@ -634,14 +634,12 @@ pub struct TokenizedBankValue1 {
     pub bank_routing_number: Option<String>,
     pub bic: Option<String>,
     pub bank_sort_code: Option<String>,
-    pub blz: Option<String>,
-    pub bank_transit_number: Option<String>,
+    pub iban: Option<String>,
```
Comment (jagan-jaya) [inline crates/api_models/src/payment_methods.rs:637]: mask it with secret

File: crates/api_models/src/payouts.rs [hunk 5]
```diff
@@ -205,110 +234,14 @@ pub struct Bank {
...
-                    bank_transit_number: p.bank_transit_number,
-                    bank_name: p.bank_name,
-                    bank_country_code: p.bank_country_code,
-                    bank_city: p.bank_city,
-                }),
-        }
-    }
+    /// [8 / 11 digits] Bank Identifier Code (bic) / Swift Code - used in many countries for identifying a bank and it's branches
+    #[schema(value_type = String, example = "HSBCGB2LXXX")]
+    pub bic: Option<String>,
```
Comment (jagan-jaya) [inline crates/api_models/src/payouts.rs:244]: mask it with secret

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

File: crates/api_models/src/payouts.rs (hunk 2 above)
Comment (kashif-m) [reply crates/api_models/src/payouts.rs:203]: Resolved in a separate commit - https://github.com/juspay/hyperswitch/pull/1384/commits/ac41daafceda596d06c428eb4b15eb15fadd337b

File: crates/api_models/src/payouts.rs (hunk 3 above)
Comment (kashif-m) [reply crates/api_models/src/payouts.rs:240]: Resolved in a separate commit - https://github.com/juspay/hyperswitch/pull/1384/commits/ac41daafceda596d06c428eb4b15eb15fadd337b

File: crates/api_models/src/payment_methods.rs (hunk 4 above)
Comment (kashif-m) [reply crates/api_models/src/payment_methods.rs:637]: Resolved in a separate commit - https://github.com/juspay/hyperswitch/pull/1384/commits/ac41daafceda596d06c428eb4b15eb15fadd337b

File: crates/api_models/src/payouts.rs (hunk 5 above)
Comment (kashif-m) [reply crates/api_models/src/payouts.rs:244]: Resolved in a separate commit - https://github.com/juspay/hyperswitch/pull/1384/commits/ac41daafceda596d06c428eb4b15eb15fadd337b

**Pull request merged**